File: Arrays&Hashing/ContainsDuplicate/Solution1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        var setUnique = new Set();
        for (let index = 0; index < nums.length; index++) {
            const element = nums[index];
            if (setUnique.has(element)) {
                return true;
            }
            setUnique.add(element);
        }
        return false;
    }
}

File: Arrays&Hashing/ContainsDuplicate/Solution1.cs
public class Solution
{
    public bool hasDuplicate(int[] nums)
    {
        var uniqueSet = new HashSet<int>();
        foreach (var num in nums)
        {
            if (!uniqueSet.Add(num))
            {
                return true; // Found a duplicate
            }
        }

        return false;
    }
}